marlowe-playground-client/src/SimulationPage/State.purs

```diff
@@ -83,68 +87,36 @@ handleAction (SetInitialSlot initialSlot) = do
 
 handleAction StartSimulation = do
   maybeInitialSlot <- peruse (_currentMarloweState <<< _executionState <<< _SimulationNotStarted <<< _initialSlot)
-  for_ maybeInitialSlot \initialSlot -> do
-    saveInitialState
-    -- Create an empty SimulationRunning in the initial slot
-    -- And then update the state in place.
-    -- TODO: This used to be done with `moveToSignificantSlot`, but it had a problem as it adds a new element in the
-    --       _marloweState, so the UNDO button was available just after starting the simulation, and when you pressed it
-    --       the Actions said: "No available actions". Now it works fine, but I don't like the fact that if you call
-    --       StartSimulation with a NEL of more than 1 element, you'd get weird behaviours. We should revisit the logic
-    --       on _oldContracts, ResetSimulation, etc.
-    assign (_currentMarloweState <<< _executionState) (emptyExecutionStateWithSlot initialSlot)
-    modifying _marloweState (map (updatePossibleActions <<< updateStateP))
-    mCurrContract <- use _currentContract
-    case mCurrContract of
-      Just currContract -> do
-        editorSetValue (show $ genericPretty currContract)
-        setOraclePrice
-      Nothing -> pure unit
+  maybeExtendedContract <- peruse (_currentMarloweState <<< _executionState <<< _SimulationNotStarted <<< _extendedContract <<< _Just)
+  for_ maybeInitialSlot \initialSlot ->
+    for_ maybeExtendedContract \extendedContract ->
+      for_ (toCore (extendedContract :: EM.Contract)) \contract -> do
```

Comment:
time for the dreaded `MaybeT` I think!

Reply:
I will try :)

Reply:
Success with a little help from @hrajchert